sdk/search/Azure.Search.Documents/tests/Utilities/SearchResources.cs

```diff
@@ -442,6 +547,90 @@ private async Task<SearchResources> CreateSearchServiceIndexAndDocumentsAsync()
             return this;
         }
 
+        /// <summary>
+        /// Creates a new storage account.
+        /// </summary>
+        /// <returns>The current <see cref="SearchResources"/>.</returns>
+        private async Task<SearchResources> CreateStorageAccountAsync()
```

Comment:
Does this have to happen per test?  Or could we do this the "normal" way with our test-resources.json and use a random container inside the storage account?

Reply:
It was done following the same pattern we do for Search services. I suppose I could create a single storage account using test-resources.json, but 1) that doesn't get run for user-driven tests, and 2) resolving that creates an inconsistency with Search service. We'd still have to create one in the F5/test run cases and only one, but could otherwise randomize containers - unless we can safely assume the container will be immutable. Thinking about that more, that problably is a safe assumption.

Was there some concern? Creating these resources is pretty fast.

Reply:
I would just prefer to avoid management plane dependencies in our tests.  We couldn't for the Search service instances.  If it's possible to do that for Storage, I would prefer to and maintain as much adherence to the unified test-resources.json approach as possible.  We'd have to create one ahead of time for local testing, but we already have to do that for the resource group/principal/etc.  We'd just have another env var with the connection string.

Reply:
We'll need to update the contrib docs to tell people to use it beforehand. I've opened #1108 to track. We should use that to create the resource group (e.g. "rg-$BaseName") and storage account.